### ISSUE TITLE
Not working with Internet explorer

### ISSUE THREAD

**Issue by ipa1981** [yuche/vue-strap#353]
https://yuche.github.io/vue-strap/ page doesn't open with neither IE version.

![image](https://cloud.githubusercontent.com/assets/15179453/18755165/be26bc70-80f3-11e6-9924-d68c774a8412.png)

I also figured our that carousel component also throws critical errors on all Internet Explorer versions. Maybe these two problems are related.

**Comment by StevenBock** [comment]
I've just run into this with the modal component as well.

**Comment by xori** [comment]
The Navbar component also throws errors

**Comment by wffranco** [comment]
vue by default says that support IE 9+, so if you are using an older version this don't work.
Anyway I don't test it because i have IE 8 and vue don't work there :v

**Comment by ipa1981** [comment]
@wffranco it doesn't work on **any** IE. Nobody talks here about IE8 :)

**Comment by wffranco** [comment]
@ipa1981 i just said the vue limitations and that I don't have any "suported" IE for testing, just in case you used some old version.
I'm downloading IE9, then I can test, right now I can't say what happen :stuck_out_tongue:

**Comment by ipa1981** [comment]
@wffranco any luck with downloading? Can you reproduce these errors?

**Comment by xori** [comment]
Just a note that it seems to be your [`NodeList`](https://github.com/yuche/vue-strap/blob/master/src/utils/NodeList.js#L229) implementation. And it throws an exception when
`this = [ window ]`.

Changing the above linked line to a traditional for loop
```javascript
for (var i = 0; i < this.length; i++) {
  var el = this[i];
```
somehow fixes the problem enough to at least paint the first screen. However other issues then surface.

**Comment by wffranco** [comment]
@ipa1981 sorry but microsoft disabled the download of older IE, in win7 I can download only IE11, and I don't to download that kind of things from other sources, can have some virus.
@xori vuestrap use webpack, this translate all the newer javascript features (ES6) to a older version (ES5), maybe the problem is that the version of IE you're using don't have a full support of ES5 too.
They say that work with IE9+, but maybe is not all truth... but work in the last version and in edge.

**Comment by ipa1981** [comment]
@wffranco problem is not only in older IE. You may try it in IE11. In my case I got error in IE11. but just reported, that the problem exists in older versions as well.

**Comment by xori** [comment]
I was using windows 8.1 IE 11.

**Comment by xori** [comment]
Yup. Just confirmed doing
```bash
npm install --save-dev babel-polyfill
```
and changing my webpack config entry to
```javascript
entry: [
    'babel-polyfill',
    path.resolve('js', 'app', 'my.app.js'),
  ],
```
makes my application and the vue-strap docs website work in IE 11.

**Comment by xori** [comment]
I mean, the vue-strap docs are still borked in IE11 (navbar dropdown not working) but it's at least showing up.

**Comment by wffranco** [comment]
ok I check this later.

**Comment by rick-hidef** [comment]
I am experiencing the same issue in IE11. Changing my webpack configuration from 
```
entry: {
    app: './src/main.js'
  },
```
to
```
entry: {
    'babel-polyfill': 'babel-polyfill',
    app: './src/main.js'
  },
```
Appears to have fixed the problem for me.

**Comment by metamet** [comment]
Is there an update to this? It seems like @xori and @rick-hidef solved this problem, yet I don't see it committed. 

Not working at all in Internet Explorer 11 renders Vue-Strap useless in real world implementations. I would highly recommend resolving this issue ASAP.

**Comment by metamet** [comment]
Created pull request: https://github.com/yuche/vue-strap/pull/384

**Comment by bruceCzK** [comment]
I'm facing the same issue in android wechat brwoser, but it doesn't throw an error, instead it cause all href link broken which is wired.

**Comment by ipa1981** [comment]
Dear @wffranco  could you please check or accept PR by community :). I agree with @metamet that it's quite urgent problem to be solved for real world applications.

**Comment by xori** [comment]
by accepting the pull request the output will increase in size by roughly 85kb. A task should probably follow to then remove this dependency in the future by converting all incompatible code. The good news is that most of the bad code is in the docs, not actually in the components.

**Comment by ipa1981** [comment]
@xori I agree plus 85kb is not the best solution.  But it's good as quickfix to make lib finally work in IE ASAP.

**Comment by wffranco** [comment]
Instead of adding the pollyfill to the main vue-strap compilation I think you can add it apart, until somebody fix the problem. Maybe explain that in the docs too... But like @xori said I think that add 85kb to the source is not the best solution.
If I can I'll check that things, but when I have a little more time, right now I'm very busy.

**Comment by ipa1981** [comment]
@wffranco I've added a polyfill locally. thanks
Would do some fixes at least in some places, if someone accepts PR here)

**Comment by wffranco** [comment]
I can accept the PR. Just try to use a clean syntax.

**Comment by ipa1981** [comment]
@wffranco can you check if I did PR correctly? This is my first time.

**Comment by wffranco** [comment]
done. and done with other sorry for the time, i'm a little bussy.
can you make the build? right now I have vue2 in my work space

**Comment by ipa1981** [comment]
@wffranco it seems something is broken after you did other changes :)

`Uncaught TypeError: Cannot read property 'forEach' of undefined(…) utils.js:17 `

Have never made builds, should I simply do `npm run build` of working version and PR it?

**Comment by pomarc** [comment]
the datepicker breaks it for me.

**Comment by wffranco** [comment]
Well I finally fix all, all tested in Edge and all fine.
I don't have any version of internet explorer, so please test it and tell me if you find some bug or is working right.

**Comment by pomarc** [comment]
https://yuche.github.io/vue-strap/ now works on IE11,.
this bug  https://github.com/yuche/vue-strap/pull/303 persists.

**Comment by ipa1981** [comment]
It works well now. It still doesn't work with IE9. But I'll make another PR for this.

**Comment by wffranco** [comment]
@pomarc I don't have time to check the datepicker in deep, but I think that is a global error, not only on IE.

**Comment by muermaru** [comment]
in ie9, the datepicker looks all right,then when I select a day but noting happend...
at the same time,the Select's clear button also has the same problem
